### DISCUSSION TITLE
Recurring events showing on wrong day.

### DISCUSSION THREAD

**Discussion by bbasselgia** [MMRIZE/MMM-CalendarExt3#146] [Q&A]
have a recurring event on my Google Calendar which is being displayed on the wrong day.
The event happens on the first Tuesday of every month at 7:00 pm ET.   
In Google Calendar it displays properly.
In CalendarExt3 it displays on Monday instead of Tuesday.
This only started happening after the last update to CalendarExt3, before that it displayed properly on Tuesdays.

**Comment by eouia** [comment]
https://github.com/MMRIZE/MMM-CalendarExt3/wiki/To-fix-wrong-repeated-fullday-event-displaying-%28MM-2.27%29

**Reply by bbasselgia** [reply]
I checked out that article.   But, this isn't a full day event, it only last an hour and is showing on the wrong day.  The time of the event is correct, it's just on the wrong day.

**Reply by eouia** [reply]
Could you send me the ics file or URL? (eouia0819@gmail.com)

**Reply by bbasselgia** [reply]
Sent the URL.

**Reply by snp66** [reply]
I’m seeing the same behavior with a Weekly Event that is occurring on Mondays but is showing on Tuesdays

**Reply by eouia** [reply]
@snp66 I need the real ics file or url which you are using, and config.js also. Share with me to diagnosis.(eouia0819@gmail.com)

**Reply by eouia** [reply]
@snp66 I think this is the same thing with [this](https://github.com/MMRIZE/MMM-CalendarExt3/discussions/146#discussioncomment-9380307). Only your case is 1 day later by your timezone. See that post.

**Comment by eouia** [comment]
<img width="1437" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/1f73f248-bb86-4326-ac01-d04e4a71debd">

I've set the System timezone to `America/New_York` and then tested your config. It shows that the issued events are displayed on Monday, not Tuesday. 
However, it is also identical to the default calendar's showing. CX3 is receiving parsed data from the default calendar, and the default calendar itself is parsing it incorrectly. (See the default calendar about 3rd Jun.)
So, the issue is lying on the default calendar module.
I already reported this symptom, but it is not fixed yet.


Waiting on fixing that bug, you can make some hotfix by yourself.
**BEFORE**
<img width="1437" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/2664c45c-8f13-4ae2-9d4d-82fc44a5c1f3">


**HOTFIX**
```js
{
  module: "MMM-CalendarExt3",
  position: "bottom_bar",
  config: {
    mode: "week",
    weekIndex: 0,
    weeksInView: 5,

    eventPayload: (payload) => {
      const affectedCalendars = ["barry", "bjshared"] // List of calendars should be fixed
      return payload.map((event) => {
        if (affectedCalendars.includes(event.calendarName) && event.recurringEvent === true) {
          return {
            ...event,
            startDate: new Date(+event.startDate).setDate(new Date(+event.startDate).getDate() + 1), // delay 1 day for start
            endDate: new Date(+event.endDate).setDate(new Date(+event.endDate).getDate() + 1), // delay 1 day for end
          };
        }
        return event
      })
    }
  }
},
```

**AFTER**
<img width="1437" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/7eb9c658-80f7-430d-b0a0-f2be56eb0251">

> You should give proper name for your issued calendars.
```js
{
  fetchInterval: 300000,
  symbol: "calendar",
  url: " ... ",
  name: "barry",
  color: "blue",
},
{
  fetchInterval: 300000,
  symbol: "calendar",
  url: " ... ",
  name: "bjshared",
  color: "green"
},
```

**Reply by snp66** [reply]
Not working for me but then again I am far from a code expert. I’m getting a syntax error on the … prior to the first event. Removing the dots removed the syntax error but then I lose the recurring events.  I’ll need to troubleshoot a bit more later.

**Reply by bayfacalendar** [reply]
This worked for me! thank you so much! 

I had issue at first where I too had syntax error, but then I traced back and made sure all my brackets were closed.

**Reply by jergrayson** [reply]
Sorry this might be a silly question.  where do i put this fix?  In MM2 config file or the MMM Calendar ext 3 config?  new to this so thanks for the help..  been a very fun project.
